Sources/ClientRuntime/Middleware/Attribute.swift

```diff
@@ -41,3 +41,22 @@ public struct Attributes {
         attributes.removeValue(forKey: key.name)
     }
 }
+
+/// A type that can be used as a type-safe property bag.
+public protocol HasAttributes: AnyObject {
```

Comment:
Is there a reason why we don't use `Attributes` directly? Either way, `Attributes` should be marked with `: HasAttributes` since it does implement the protocol.

Reply:
Yea, so the reason is that I want interceptor implementations to be able to specify a more specific thing that 'has attributes', which may have helper functions for getting specific attributes. For example, `HttpContext` is now a `HasAttributes`, so in `HttpInterceptor`s, you can say 
```
let authSchemeResolver = context.getAttributes().getAuthSchemeResolver()
```
instead of
```
let authSchemeResolver = context.getAttributes().get(key: AttributeKeys.authSchemeResolver)
```
This makes it easier to adapt middleware that uses `HttpContext` to interceptors.

Reply:
Also, `Attributes` doesn't implement `HasAttributes` because `HasAttributes` can only be implemented by a reference type, i.e. a class, which is why it has the `: AnyObject`. The reason I did this was because interceptors need to mutate the attributes, and if attributes isn't a reference type, you'd need to do
```
var attrs = context.getAttributes()
attrs.set(...)
context.updateAttributes(attrs)
```